File: FE-RESTAURANT/src/interfaces/ComboResponseDTO.ts
import type { ComboDishResponseDTO } from "./ComboDishResponseDTO";
import { ItemAvailabilityStatus } from "./enums/MenuItemStatus";

export interface ComboResponseDTO {
  id: string;
  name: string;
  image: string;
  description: string;
  basePrice: number;
  availabilityStatus: ItemAvailabilityStatus;
  manualAvailabilityOverride: boolean;
  createdBy: number;
  createdAt: string;
  updatedAt: string;
  comboDishes: ComboDishResponseDTO[];
  inUse?: boolean;

  // Discount entity relationship
  discount?: {
    id: number;
    code: string;
    name: string;
    newPrice: number;
    startDate: string;
    endDate: string;
    status: string;
    description?: string;
  };
} 

File: RESTAURANT/src/main/java/com/poly/restaurant/dtos/ComboResponseDTO.java
package com.poly.restaurant.dtos;

import com.poly.restaurant.entities.enums.ItemAvailabilityStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ComboResponseDTO {
    private Long id;
    private String name;
    private String image;
    private String description;
    private BigDecimal basePrice;
    private ItemAvailabilityStatus availabilityStatus;
    private Boolean manualAvailabilityOverride;
    // Discount fields
    private BigDecimal discountPercentage;
    private BigDecimal discountAmount;
    private LocalDateTime discountStartDate;
    private LocalDateTime discountEndDate;
    private Boolean discountActive;
    private BigDecimal finalPrice; // Giá cuối cùng sau khi áp dụng giảm giá
    private Long branchId;
    private String branchName;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<ComboDishResponseDTO> comboDishes;
    private Boolean inUse;
}